api/app/sfms/hourly_ffmc_processor.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import os
+import tempfile
+from datetime import datetime, timedelta
+from osgeo import gdal
+from typing import List, cast
+
+from app.weather_models.rdps_filename_marshaller import model_run_for_hour
+
+from app.geospatial.wps_dataset import WPSDataset
+from app.jobs.rdps_sfms import MAX_MODEL_RUN_HOUR
+from app.sfms.daily_fwi_processor import MultiDatasetContext
+from app.sfms.fwi_processor import calculate_ffmc
+from app.sfms.raster_addresser import RasterKeyAddresser
+from app.utils.geospatial import GDALResamplingMethod
+from app.utils.s3 import set_s3_gdal_config
+from app.utils.s3_client import S3Client
+
+
+logger = logging.getLogger(__name__)
+
+
+class HourlyFFMCProcessor:
+    """
+    Class for calculating/generating forecasted hourly FFMC rasters.
+    """
+
+    def __init__(self, start_datetime: datetime, addresser: RasterKeyAddresser):
+        self.start_datetime = start_datetime
+        self.addresser = addresser
+
+    async def process(self, s3_client: S3Client, input_dataset_context: MultiDatasetContext, hours_to_process: int = MAX_MODEL_RUN_HOUR):
+        set_s3_gdal_config()
+
+        # 1 - Determine starting hFFMC (4am or 4pm) from SFMS and get key, confirm exists, if not, exit
+        # 2 - Determine what would be last key of run and check if exists, if exists, exit
+        # 3 - Get all weather variable keys and check if last one exists, if not, exit
+        # 4 - Use seed hFFMC plus:
+        #       - rh, temp and wind speed from RDPS model run hour n = 000
+        #       - computed precip at n = 0Z or 12Z
+        # 5 - Use newly calculated hFFMC to calculate next hFFMC using:
+        #       -  rh, temp and wind speed from RDPS model run hour n + 1
+        #       - computed precip at n + 1
+        # hFFMC files from SFMS use PST datetimes
```

Comment:
I think our SFMS uploads use whatever local time is in Vancouver's time zone, so during the fire season these would be uploaded in PDT (see app/utils/sfms.py). I think your still grabbing the correct input data though, if my mental timezone converting is correct

Reply:
Fixed in https://github.com/bcgov/wps/pull/4153/commits/0e7bda3053128b3150a09055b2964353eb2a82a9